src/main/java/org/openhab/tools/analysis/checkstyle/EshInfXmlCheck.java

```diff
@@ -0,0 +1,348 @@
+    public static final String THING_DIRECTORY = "thing";
+    public static final String BINDING_DIRECTORY = "binding";
+    public static final String ESH_INF_DIRECTORY = "ESH-INF";
+    public static final String CONFIGURATION_DIRECTORY = "config";
+
+    private static final String CONFIG_DESCRIPTION_EXPRESSION = "//config-description[@uri]/@uri";
+    private static final String CONFIG_DESCRIPTION_REF_EXPRESSION = "//config-description-ref[@uri]/@uri";
+    private static final String BRIDGE_TYPE_EXPRESSION = "//bridge-type[@id]/@id";
+    private static final String SUPPORTED_BRIDGE_TYPE_REF_EXPRESSION = "//supported-bridge-type-refs/bridge-type-ref[@id]/@id";
+
+    public static final String MESSAGE_MISSING_URI_CONFIGURATION = "Missing configuration for the configuration reference with uri - {0}";
+    public static final String MESSAGE_MISSING_SUPPORTED_BRIDGE = "Missing the supported bridge with id {0}";
+    public static final String MESSAGE_UNUSED_URI_CONFIGURATION = "Unused configuration reference with uri - {0}";
+    public static final String MESSAGE_UNUSED_BRIDGE = "Unused bridge reference with id - {0}";
+    public static final String MESSAGE_EMPTY_FILE = "The file {0} should not be empty.";
+    private static final String MESSAGE_NOT_INCLUDED_XML_FILE = "The file {0} isn`t included in the build.properties file. Good approach is to add include all files by adding `ESH-INF/` value to the bin.includes property.";
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private Map<String, File> allConfigDescriptionRefs;
+    private Map<String, File> allConfigDescriptions;
+
+    private Map<String, File> allSupportedBridges;
+    private Map<String, File> allBridgeTypes;
+
+    private Map<Path, File> eshInfFiles;
+    private IBuild buildPropertiesFile;
+
+    private String thingSchema;
+    private String bindingSchema;
+    private String configSchema;
+
+    // A configuration property for schema file.
+    public void setThingSchema(String thingSchema) {
+        this.thingSchema = thingSchema;
+    }
+
+    // A configuration property for schema file.
+    public void setBindingSchema(String bindingSchema) {
+        this.bindingSchema = bindingSchema;
+    }
+
+    // A configuration property for schema file.
+    public void setConfigSchema(String configSchema) {
+        this.configSchema = configSchema;
+    }
+
+    public EshInfXmlCheck() {
+        setFileExtensions(THING_TYPE_EXTENSION, PROPERTIES_EXTENSTION);
+
+        eshInfFiles = new HashMap<>();
+
+        allConfigDescriptionRefs = new HashMap<>();
+        allConfigDescriptions = new HashMap<>();
+
+        allSupportedBridges = new HashMap<>();
+        allBridgeTypes = new HashMap<>();
+    }
+
+    @Override
+    public void beginProcessing(String charset) {
+        logger.debug("Executing the {}", this.getClass().getSimpleName());
```

Comment:
Will X also show trace? I already saw mentions that the output exploded when using the -X, so that why mentioned it. Maybe then just leave it out?